src/System.Management.Automation/engine/MshCommandRuntime.cs

```diff
@@ -3018,18 +3123,34 @@ internal ActionPreference WarningPreference
             {
                 // Setting CommonParameters.WarningAction has highest priority
                 if (IsWarningActionSet)
+                {
                     return _warningPreference;
+                }
 
                 if (Debug)
+                {
                     return ActionPreference.Inquire;
+                }
+
                 if (Verbose)
+                {
                     return ActionPreference.Continue;
+                }
+
```

Comment:
Another similar question for reviewers:

Why are lines 3130-3139 here? We're handling the warning action preference, but we're checking `Debug` and `Verbose` in the process? Logically this made no sense to me and I wanted to remove those lines completely, but I don't have any history behind this code and why that logic is present here, so I wanted to ask others.

Reply:
I figured this out, and updated the code accordingly with comments plus a minor update. See other review comments or the code comments that will be added in next commit where I explain how this works if you're interested in the details.